Doors are devices

xrmx opened the pull request:
This is done on top of #6 in order to not create conflicts.

xrmx committed:
labAdmin: cleanup LogDevice stop

While at it add a test

xrmx committed:
labAdmin: show device last activity in admin

Fix #26

xrmx committed:
labadmin: make some sense of LogDevice.priceWork

And add a test

xrmx committed:
labAdmin: move token extraction from request into an helper

As it will be handy for views needing a device.

xrmx committed:
labadmin: add view to start logging the use of a device

It takes the user card nfc id as nfc_id through a POST request.

Ref #26

xrmx committed:
labAdmin: fix UserProfile.can_use_device_now

Like it actually works and doesn't silently fail under a catch all
try block.

xrmx committed:
labAdmin: add view to stop logging the use of a device

It takes the user card nfc id as nfc_id through a POST request.

Ref #26

xrmx committed:
labAdmin: remove unused functions

And remove duplicated import while at it.

xrmx committed:
labAdmin: kill role kind

We now treats doors as devices which means that we can kill the
role_kind and the various can_use_door_now methods.

So now OpenDoorByNFC requires the the token authentication as
the credits API.

xrmx committed:
labAdmin: kill unused can_use_device_now

We don't need them and probably they aren't working either.

tagaf commented:
The devices as have been implemented till now can parse JSON. So there's no problem in giving error 201 when access is negated.

xrmx committed:
labAdmin: remove unused data from tests fixture

xrmx committed:
labAdmin: add tests for UserProfile.can_use_device_now()

xrmx commented:
@tagaf 👍

xrmx committed:
labAdmin: remove commented routes in urls

xrmx committed:
labAdmin: check categories in can_use_device_now

So that we can implement per category of device permission and
not just for roles.
